xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -94,12 +119,15 @@ public static XdsServerBuilder forPort(int port, ServerCredentials serverCredent
   @Override
   public Server build() {
     checkState(isServerBuilt.compareAndSet(false, true), "Server already built!");
-    AtomicReference<FilterChainSelector> filterChainSelectorRef = new AtomicReference<>();
-    InternalNettyServerBuilder.eagAttributes(delegate, Attributes.newBuilder()
-            .set(ATTR_FILTER_CHAIN_SELECTOR_REF, filterChainSelectorRef)
-            .build());
+    FilterChainSelectorManager filterChainSelectorManager = new FilterChainSelectorManager();
+    Attributes.Builder builder = Attributes.newBuilder()
+            .set(ATTR_FILTER_CHAIN_SELECTOR_MANAGER, filterChainSelectorManager);
+    if (drainGraceTimeUnit != null) {
+      builder.set(ATTR_DRAIN_GRACE_NANOS, drainGraceTimeUnit.toNanos(drainGraceTime));
```

Comment:
Can we make `drainGraceTime` a property of the `FilterChainSelectorManager`? We don't really need a separate attribute `ATTR_DRAIN_GRACE_NANOS` to pass the value in that case.

Reply:
Yeah, I saw that when implementing, but it seemed like false coupling. If anything, I'd create a new class to hold the two values and put that class in the attributes. But this is fine for now.